src/Table.hxx

```diff
@@ -306,9 +378,89 @@ public:
 
     void shrink_ipac_string_columns_to_fit(const std::vector<size_t> &column_widths);
 
+    std::string extract_value_as_string(const std::string &col_name,
+                                        size_t row_id) const;
+    std::string extract_value_as_string(size_t col_id, size_t row_id) const;
+
     std::vector<std::string> extract_column_values_as_strings(
             const std::string &colname) const;
 
+    template <typename T>
+    std::vector<T> extract_value(const std::string &col_name, size_t row_id) {
+        auto col_id = column_index(col_name);
+        return extract_value<T>(col_id, row_id);
+    }
+
+    template <typename T>
+    std::vector<T> extract_value(size_t col_id, size_t row_id) {
+        std::vector<T> val_array;
+        extract_value(val_array, col_id, row_id);
+        return val_array;
+    }
+
+    template <typename T>
+    void extract_value(std::vector<T> &val_array, size_t col_id, size_t row_id) {
+        if (col_id >= columns.size()) {
+            throw std::runtime_error("Invalid column index: " + std::to_string(col_id));
+        }
+        if (row_id >= num_rows()) {
+            throw std::runtime_error("Invalid row index: " + std::to_string(row_id));
+        }
+        if (std::is_same<T, char>::value) {
+            std::string msg(
+                    "extract_value() is not supported for columns of type char; ");
+            msg.append("please use extract_value_as_string().");
+            throw std::runtime_error(msg);
+        }
```

Comment:
This should be a compile time error, e.g.

```c++
    static_assert(!std::is_same<T, char>::value,
                  "Table::extract_value() is not supported for columns of type char")
```

Alternatively, I think you could say something like:

```c++
template <typename T,
          typename = std::enable_if_t<!std::is_same<T, char>::value>>
void extract_value(...) { ... }
```

to remove the `char` specialization.